subprojects/signing/src/main/java/org/gradle/plugins/signing/SigningExtension.java

```diff
@@ -305,7 +311,75 @@ public void execute(Sign task) {
         return result;
     }
 
-    private Sign createSignTaskFor(final CharSequence name, Action<Sign> taskConfiguration) {
+    /**
+     * Creates signing tasks that sign all publishable artifacts of the given publications.
+     *
+     * <p>The created tasks will be named "sign<i>&lt;publication name capitalized&gt;</i>Publication".
+     * That is, given a publication with the name "mavenJava" the created task will be named "signMavenJavaPublication".
+     *
+     * The signature artifacts for the created tasks are added to the publishable artifacts of the given publications.
+     *
+     * @param publications The publications whose artifacts are to be signed
+     * @return the created tasks.
+     * @since 4.8
+     */
+    @Incubating
+    public List<Sign> sign(Publication... publications) {
```

Comment:
All other `sign` methods that take multiple arguments (e.g. configurations or tasks) also use `List` even though they could return a set because task creation would fail if there were duplicates. So, I'd vote for returning a `List` for the sake of consistency.

@oehme Thoughts?

Reply:
I guess it's not a big deal, though things like `contains` would be slower even though we could support it quickly.